src/pkg/util/nvidiautils/nvidiautils.go

```diff
@@ -0,0 +1,159 @@
+				strArray = append(strArray, newentry[0]) // add prefix (filepath.so)
+			}
+		}
+	}
+	return strArray, bindArray, nil
+}
+
+// generate bind list using contents of nvliblist.conf
+func nvidiaLiblist(abspath string) ([]string, error) {
+	var strArray []string
+
+	// grab the entries in nvliblist.conf file
+	file, err := os.Open(abspath + "/nvliblist.conf")
+	if err != nil {
+		return nil, fmt.Errorf("%v", err)
+	}
+
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "#") && line != "" {
+			strArray = append(strArray, line)
+		}
+	}
+	return strArray, nil
+}
+
+// GetNvidiaBindPath returns a string array consisting of filepaths of nvidia
+// related files to be added to the BindPaths
+func GetNvidiaBindPath(abspath string) ([]string, error) {
+	var strArray []string
+	var bindArray []string
+
+	// use nvidia-container-cli if presenet
+	strArray, bindArray, err := nvidiaContainerCli()
+	if err != nil {
+		sylog.Verbosef("nvidiaContainercli returned: %v", err)
+		sylog.Verbosef("Falling back to nvliblist.conf")
+
+		// nvidia-container-cli not present or errored out
+		// fallback is to use nvliblist.conf
+		strArray, err = nvidiaLiblist(abspath)
+		if err != nil {
+			sylog.Warningf("nvidiaLiblist returned: %v", err)
+			return nil, err
+		}
+	}
+
+	// walk thru the ldconfig output and add entries which contain the filenames
+	// returned by nvidia-container-cli OR the nvliblist.conf file contents
+	command, err := exec.LookPath("ldconfig")
+	if err != nil {
+		sylog.Warningf("ldconfig not found: %v", err)
+		return nil, nil
+	}
+
+	cmd := exec.Command(command, "-p")
+	out, err := cmd.Output()
+	if err != nil {
+		sylog.Warningf("ldconfig execution error: %v", err)
+		return nil, nil
+	}
+
+	lastadd := ""
+	for _, ldconfigOutputline := range strings.Split(strings.TrimSuffix(string(out), "\n"), "\n") {
+		if ldconfigOutputline != "" {
+			for _, nvidiaFileName := range strArray {
+
+				// sample ldconfig -p output (ldconfigOutputline)
+				// 	libnvidia-ml.so.1 (libc6,x86-64) => /usr/lib64/nvidia/libnvidia-ml.so.1
+				//	libnvidia-ml.so (libc6,x86-64) => /usr/lib64/nvidia/libnvidia-ml.so
+
+				ldconfigOutputSplitline := strings.SplitN(ldconfigOutputline, "=> ", 2)
+				if len(ldconfigOutputSplitline) > 1 {
+
+					// ldconfigOutputSplitline[0] is the "libnvidia-ml.so[.1] (libc6,x86-64)"" (from the above example)
+					// ldconfigOutputSplitline[1] is the "/usr/lib64/nvidia/libnvidia-ml.so[.1]" (from the above example)
+					// these next 2 lines extract the "libnvdia-ml.so[.1]" (from the above example)
+
+					// ldconfigFileName is "libnvidia-ml.so[.1]" (from the above example)
+					ldconfigFileNames := strings.Split(ldconfigOutputSplitline[0], " ")
+					ldconfigFileName := strings.TrimSpace(string(ldconfigFileNames[0]))
+
+					if strings.HasPrefix(ldconfigFileName, nvidiaFileName) && ldconfigFileName != lastadd { // add if not duplicate
```

Comment:
I think the addition of `strings.HasPrefix()` will pick up any `*.so.1` libs or others with different suffixes.  But needs to be tested to be sure.